server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java

```diff
@@ -568,6 +575,45 @@ public GeoPoint normalizeFromSource(GeoPoint point) {
             // normalize during parsing
             return point;
         }
+
+        @Override
+        protected void parseAndConsumeFromObject(
+            XContentParser parser,
+            CheckedConsumer<GeoPoint, IOException> consumer,
+            MalformedValueHandler malformedHandler
+        ) throws IOException {
+            XContentParser parserWithCustomization = parser;
+            XContentBuilder malformedDataForSyntheticSource = null;
+
+            if (storeMalformedDataForSyntheticSource) {
+                if (parser.currentToken() == XContentParser.Token.START_OBJECT
+                    || parser.currentToken() == XContentParser.Token.START_ARRAY) {
+                    // We have a complex structure so we'll memorize it while parsing.
+                    var memorizingParser = new MemorizingXContentParser(parser);
```

Comment:
Nit: assign directly to parserWithCustomization?

Reply:
`parserWithCustomization` and `memorizingParser` are different types.